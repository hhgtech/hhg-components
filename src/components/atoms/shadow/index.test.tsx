import React from 'react'

import { render, cleanup } from '@testing-library/react'
import 'jest-styled-components'

import { Shadow } from '.'
import MockTheme from '../../../misc/mockTheme'

afterEach(cleanup)

test('it renders', () => {
  const shadow = render(
    <MockTheme>
      <Shadow size="medium">
        <div>Shadow child!</div>
      </Shadow>
    </MockTheme>,
  )

  const self = shadow.container.firstChild

  expect(self).toMatchSnapshot()

  expect(self).toHaveAttribute('data-shadow-size', 'medium')
})
