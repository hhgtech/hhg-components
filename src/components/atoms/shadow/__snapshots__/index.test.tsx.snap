// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
.c0[data-shadow-size='small'] {
  -webkit-filter: drop-shadow(0px 4px 8px rgba(0,0,0,0.04)) drop-shadow(0px 0px 2px rgba(0,0,0,0.06)) drop-shadow(0px 0px 1px rgba(0,0,0,0.04));
  filter: drop-shadow(0px 4px 8px rgba(0,0,0,0.04)) drop-shadow(0px 0px 2px rgba(0,0,0,0.06)) drop-shadow(0px 0px 1px rgba(0,0,0,0.04));
}

.c0[data-shadow-size='normal'] {
  -webkit-filter: drop-shadow(0px 10px 16px rgba(0,0,0,0.04)) drop-shadow(0px 2px 8px rgba(0,0,0,0.04)) drop-shadow(0px 0px 1px rgba(0,0,0,0.04));
  filter: drop-shadow(0px 10px 16px rgba(0,0,0,0.04)) drop-shadow(0px 2px 8px rgba(0,0,0,0.04)) drop-shadow(0px 0px 1px rgba(0,0,0,0.04));
}

.c0[data-shadow-size='medium'] {
  -webkit-filter: drop-shadow(0px 16px 24px rgba(0,0,0,0.06)) drop-shadow(0px 2px 8px rgba(0,0,0,0.04)) drop-shadow(0px 0px 1px rgba(0,0,0,0.04));
  filter: drop-shadow(0px 16px 24px rgba(0,0,0,0.06)) drop-shadow(0px 2px 8px rgba(0,0,0,0.04)) drop-shadow(0px 0px 1px rgba(0,0,0,0.04));
}

.c0[data-shadow-size='large'] {
  -webkit-filter: drop-shadow(0px 24px 32px rgba(0,0,0,0.04)) drop-shadow(0px 16px 24px rgba(0,0,0,0.04)) drop-shadow(0px 4px 8px rgba(0,0,0,0.04)) drop-shadow(0px 0px 1px rgba(0,0,0,0.04));
  filter: drop-shadow(0px 24px 32px rgba(0,0,0,0.04)) drop-shadow(0px 16px 24px rgba(0,0,0,0.04)) drop-shadow(0px 4px 8px rgba(0,0,0,0.04)) drop-shadow(0px 0px 1px rgba(0,0,0,0.04));
}

.c0[data-shadow-size='soft'] {
  -webkit-filter: drop-shadow(0px 24px 32px rgba(0,0,0,0.04)) drop-shadow(0px 16px 24px rgba(0,0,0,0.04)) drop-shadow(0px 4px 8px rgba(0,0,0,0.04)) drop-shadow(0px 0px 1px rgba(0,0,0,0.04));
  filter: drop-shadow(0px 24px 32px rgba(0,0,0,0.04)) drop-shadow(0px 16px 24px rgba(0,0,0,0.04)) drop-shadow(0px 4px 8px rgba(0,0,0,0.04)) drop-shadow(0px 0px 1px rgba(0,0,0,0.04));
}

<div
  class="c0"
  data-shadow-size="medium"
>
  <div>
    Shadow child!
  </div>
</div>
`;
